pkg/component/component.go

```diff
@@ -0,0 +1,229 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package component
+
+import (
+	"fmt"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+)
+
+// Unit is a single unit that needs to be running inside a component.
+type Unit struct {
+	ID     string
+	Type   client.UnitType
+	Config map[string]interface{}
+}
+
+// Component is a set of units that needs to run.
+type Component struct {
+	Spec  InputRuntimeSpec
+	Units []Unit
+}
+
+// ToComponents returns the components that should be running based on the policy and the current runtime specification.
+func (r *RuntimeSpecs) ToComponents(policy map[string]interface{}) ([]Component, error) {
+	outputsMap, err := toIntermediate(policy)
+	if err != nil {
+		return nil, err
+	}
+	if outputsMap == nil {
+		return nil, nil
+	}
```

Comment:
How that return is possible, is this a guard for an empty agent policy?

Reply:
Yes we consider an empty policy valid, the agent would just go dormant waiting on a policy revision that has outputs and inputs.

Reply:
I see, thanks for the explaination.